src/client/sandbox/event/message.js

```diff
@@ -169,12 +169,16 @@ export default class MessageSandbox extends SandboxBase {
         if (!this._isIFrameRemoved() && (isIFrameWithoutSrc || !isCrossDomainWindows(targetWindow, this.window) &&
                                                                targetWindow[this.RECEIVE_MSG_FN])) {
             //NOTE: postMessage delay imitation
-            nativeMethods.setTimeout.call(this.topWindow, () =>
+            nativeMethods.setTimeout.call(this.topWindow, () => {
+                // NOTE: GH-171
+                if (!isSafari || this.window.top) {
```

Comment:
We want to leave the opportunity to communicate with a removed frame. This may be necessary in some cases. Unfortunately, we can not implement it in the Safari.